components/tel/Core/Src/radio.c

```diff
@@ -65,13 +73,59 @@ static void copy_CAN_id(uint8_t* source, CAN_Radio_msg_t *tx_CAN_msg, uint32_t c
 
 
 /**
- * @brief Sends a CAN message over the radio by packing it into a CAN_BUFFER_LEN buffer 
- *        and transmits it over UART
+ * @brief Takes api_buffer and packages it into an api frame. transmits this api frame over uart.
+ * @param radio_buffer[]: pointer to an array in which the messages are being added, api_buffer_postion: position in api_buffer where messages end.
+ * @return void
+*/
+void RADIO_tx_API_Packager(uint8_t api_buffer[], uint16_t api_buffer_position)
+{
+ uint16_t api_packet_size = API_OVERHEAD_SIZE + api_buffer_position;
+ uint8_t api_packet[api_packet_size];
+ XBEE_api_overhead_setup(api_packet);
+
+ for (int i = 0; i<api_buffer_position; i++){
+	 api_packet[MESSAGE_DATA_START_POSITION + i] = api_buffer[i];
```

Comment:
Message data start position constant should be 18 not 19. This is because 18 bytes of overhead corresponds to index 17. So the first index of the first message would be the next byte which is index 18 (but byte 19).

Reply:
Good catch - I think I forgot to add the bytes of overhead that we add to the message (can byte and number of messages) into the overhead bytes count (and the rest of the code so I will have to implement that). Overall, there are 20 bytes of overhead. One byte, the checksum, comes after the message_data start position. So the data position should be 18.